primal-check/Cargo.toml

```
@@ -15,10 +15,12 @@ Fast standalone primality testing.
 """
 
 [dependencies]
-num-integer = "0.1"
+libm = { version = "0.2.8", optional = true }
+num-integer = { version = "0.1", default_features = false }
 
 [dev-dependencies]
 primal = { path = "..", version = "0.3" }
 
 [features]
+no-std = ["libm"]
```

I think we should bump the MSRV to 1.60 and use the new syntax, to avoid an implicit `feature = "libm"`.
```suggestion
no-std = ["dep:libm"]
```

---

stupid question, what do you mean adjust the MSRV. I have a vague idea that yearly editions correspond to rust releases but I don't know the mapping

---

We have a minimum version of the Rust compiler that we support -- that's the MSRV, nowadays represented in [the `rust-version` field](https://doc.rust-lang.org/cargo/reference/manifest.html#the-rust-version-field). We haven't marked that explicitly yet, but at the moment we're testing with Rust 1.36. If we use this `dep:` syntax, that only works with Rust 1.60 and later.

If we instead go with a semver bump and use additive `libm` and `std` features, then I think we don't need `dep:`, but I think then we might as well go with 1.56 and upgrade to 2021 edition.

---

I don't know of anyone with hard constraints on `primal` compatibility, so it's mostly just going to be what's convenient to us, or not too much burden to support.

---

if making things purely additive, should features `libm` and `std` be mutually exclusive?

---

Nah, we can just decide that `std` "wins". There's no semantic effect where we need to force only one or the other, like you might with something like async runtimes; the only harm is that `libm` will go unused.

---

The `std` implementation of these functions should be the "best" (fastest, most accurate, etc.) available on any given target, whereas the `libm` crate is more of a fallback solution.